### DISCUSSION TITLE
openMVG vs ORBSLAM3

### DISCUSSION THREAD

**Discussion by navid-mahmoudian** [openMVG/openMVG#2095] [Q&A]
Hello,

My apologies for my naive question. Would anyone be able to explain to me the differences, pros and cons of using openMVG over [ORBSLAM3](https://github.com/UZ-SLAMLab/ORB_SLAM3)?

**Comment by pmoulon** [comment]
ORBSLAM3 is a SLAM library => SLAM performs 3d recovery of the camera motion and some scene points using videos

OpenMVG is performing SFM (Structure From Motion) -> recovery of the camera motion and some scene points for unstructured image datasets (photos of a church, ...)

**Reply by navid-mahmoudian** [reply]
Thank you very much for your response. This is the general definition of SLAM vs SFM. I am asking more about the capabilities of each library. For example, could you please answer the following questions:

1- orbslam3 supports RGB-D data. Is it possible to do sfm with RGB-D images in openMVG?

2- The other question is that orbslam3 supports imu data and since these types of data are noisy, they do a joint optimization (if I am not mistaken) to improve the results using all kinds of information. IMU data is also useful to solve scale problem of sfm for 3d reconstruction. I know in OpenMVG it is possible to import gps data, but does it do the joint optimization? Does it solve the scale problem?

3- in terms of algorithmic procedure, can we say SLAM is equivalent to incremental (sequential) SFM?

**Reply by pmoulon** [reply]
1- OpenMVG does not support RGB-D data
2- OpenMVG does not support IMU inputs
3- Not totally, since incremental SfM and BA consider the entire view graph (so handle loop closure naturally). SLAM often minimize reprojection error on local windows of images, and so does not natively support loop closures.

**Comment by navid-mahmoudian** [comment]
Thank you very much for your response and this amazing tool you have provided.

If I want to work with RGB-D images on openMVG, which part of the code do you think must be modified in openMVG? I guess the only file that must be changed in the pipeline is openMVG_main_SfM. Am I right? if yes, do you have an idea where I have to change?

Thanks again for this amazing library !
